jr-test-module/pom.xml

```diff
@@ -0,0 +1,97 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <!-- This module was also published with a richer model, Gradle metadata,  -->
+    <!-- which should be used instead. Do not delete the following line which  -->
+    <!-- is to indicate to Gradle or any Gradle module metadata file consumer  -->
+    <!-- that they should prefer consuming it instead. -->
+    <!-- do_not_remove: published-with-gradle-metadata -->
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.fasterxml.jackson.jr</groupId>
+        <artifactId>jackson-jr-parent</artifactId>
+        <version>2.17.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>jackson-jr-test-module</artifactId>
+    <packaging>bundle</packaging>
+    <description>Test Module with higher JDK Version + Groovy support to test Jackson-Jr</description>
+    <url>https://github.com/FasterXML/jackson-jr</url>
+
+    <licenses>
+        <license>
+            <name>The Apache Software License, Version 2.0</name>
+            <url>https://www.apache.org/licenses/LICENSE-2.0.txt</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+    <properties>
+        <!--    To Test Latest Java Features Lets use this settings    -->
+        <maven.compiler.source>21</maven.compiler.source>
```

Comment:
Should we use 17 instead, as that has Record type? Or is there some reason to limit to later JDKs?